tests/test_statemachine_bounded_transitions.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+
+import mock
+import pytest
+
+from statemachine import StateMachine, State
+
+
+class MyModel(object):
+    def __init__(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+    def __repr__(self):
+        return "{}({!r})".format(type(self).__name__, self.__dict__)
+
+
+class CampaignMachine(StateMachine):
+    draft = State("Draft", initial=True)
+    producing = State("Being produced")
+    closed = State("Closed")
+
+    add_job = draft.to(draft) | producing.to(producing)
+    produce = draft.to(producing)
+    deliver = producing.to(closed)
+
+    def on_enter_producing(self, **kwargs):
+        pass
```

Comment:
can you write a test that guarantee backwards compatibility? 
- The new code with a `on_enter` that is written without expecting to receive `args` or `kwargs`.

Reply:
added! I'm not sure if the way I tested by using mock will be enough :thinking:  , should I not mock these methods?